testing/integration/otel_test.go

```diff
@@ -0,0 +1,295 @@
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/require"
+
+	aTesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/estools"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/testcontext"
+	"github.com/elastic/go-elasticsearch/v8"
+)
+
+const fileProcessingFilename = `/tmp/testfileprocessing.json`
+
+var fileProcessingConfig = []byte(`receivers:
+  filelog:
+    include: [ "/var/log/system.log", "/var/log/syslog"  ]
+    start_at: beginning
+
+exporters:
+  file:
+    path: ` + fileProcessingFilename + `
+service:
+  pipelines:
+    logs:
+      receivers: [filelog]
+      exporters:
+        - file`)
+
+var apmProcessingContent = `2023-06-19 05:20:50 ERROR This is a test error message
+2023-06-20 12:50:00 DEBUG This is a test debug message 2
+2023-06-20 12:51:00 DEBUG This is a test debug message 3
+2023-06-20 12:52:00 DEBUG This is a test debug message 4`
+
+var apmOtelConfig = `receivers:
+  filelog:
+    include: [ %s ]
+    operators:
+      - type: regex_parser
+        regex: '^(?P<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}) (?P<sev>[A-Z]*) (?P<msg>.*)$'
+        timestamp:
+          parse_from: attributes.time
+          layout: '%%Y-%%m-%%d %%H:%%M:%%S'
+        severity:
+          parse_from: attributes.sev
+
+processors:
+  resource:
+    attributes:
+    # APM Server will use service.name for data stream name: logs-apm.app.<service_name>-default
+    - key: service.name
+      action: insert
+      value: elastic-otel-test
+    - key: host.test-id
+      action: insert
+      value: %s
+
+exporters:
+  debug:
+    verbosity: detailed
+    sampling_initial: 10000
+    sampling_thereafter: 10000
+  otlp/elastic:
+      endpoint: "127.0.0.1:8200"
+      tls:
+        insecure: true
+
+service:
+  pipelines:
+    logs:
+      receivers: [filelog]
+      processors: [resource]
+      exporters:
+        - debug
+        - otlp/elastic`
+
+func TestFileProcessing(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Group: Default,
+		Local: true,
+		OS: []define.OS{
+			// input path missing on windows
+			{Type: define.Linux},
+			{Type: define.Darwin},
+		},
+	})
+
+	t.Cleanup(func() {
+		_ = os.Remove(fileProcessingFilename)
+	})
+
+	fixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+	err = fixture.Prepare(ctx, fakeComponent, fakeShipper)
+	require.NoError(t, err)
+
+	// replace default elastic-agent.yml with otel config
+	// otel mode should be detected automatically
+	err = fixture.Configure(ctx, fileProcessingConfig)
+	require.NoError(t, err)
+
+	var fixtureWg sync.WaitGroup
+	fixtureWg.Add(1)
+	go func() {
+		err = fixture.RunWithClient(ctx, false)
+		fixtureWg.Done()
+	}()
```

Comment:
Nit: safety from panics ?
```suggestion
	go func() {
		defer fixtureWg.Done()
		err = fixture.RunWithClient(ctx, false)
	}()
```